src/test/scala/chiselTests/QueueFlushSpec.scala

```diff
@@ -0,0 +1,221 @@
+package chiselTests
+
+import org.scalacheck._
+
+import chisel3._
+import chisel3.testers.{BasicTester, TesterDriver}
+import chisel3.util._
+import chisel3.util.random.LFSR
+
+/** Test elements can be enqueued and dequeued
+ * 
+  * @param elements The sequence of elements used in the queue
+  * @param queueDepth The max number of entries in the queue
+  * @param bitWidth Integer size of the data type used in the queue
+  * @param tap Integer tap('seed') for the LFSR
+  * @param useSyncReadMem True uses SyncReadMem instead of Mem as an internal memory element
+  */
+class ThingsPassThroughFlushQueueTester(elements: Seq[Int], queueDepth: Int, bitWidth: Int, tap: Int, useSyncReadMem: Boolean) extends BasicTester {
+  
+  val q = Module(new Queue(UInt(bitWidth.W), queueDepth, hasFlush = true))  
+  q.io.flush.get := false.B
+  val elems = VecInit(elements.map(_.U))
+
+  val inCnt = Counter(elements.length + 1)
+  val outCnt = Counter(elements.length + 1)
+
+  q.io.enq.valid := (inCnt.value < elements.length.U)
+  q.io.deq.ready := LFSR(16)(tap)
+
+  q.io.enq.bits := elems(inCnt.value)
+  when(q.io.enq.fire()) {
+    inCnt.inc()
+  }
+  when(q.io.deq.fire()) {
+    //ensure that what comes out is what comes in
+    assert(elems(outCnt.value) === q.io.deq.bits)
+    outCnt.inc()
+  }
+  when(outCnt.value === elements.length.U) {
+    stop()
+  }
+}
+
+/** Test queue can flush at random times
+ * 
+  * @param elements The sequence of elements used in the queue
+  * @param queueDepth The max number of entries in the queue
+  * @param bitWidth Integer size of the data type used in the queue
+  * @param tap Integer tap('seed') for the LFSR
+  * @param useSyncReadMem True uses SyncReadMem instead of Mem as an internal memory element
+  */
+class QueueGetsFlushedTester (elements: Seq[Int], queueDepth: Int, bitWidth: Int, tap: Int, useSyncReadMem: Boolean) extends BasicTester {
```

Comment:
why isn't this test checking that we are still enqueueing and dequeuing the elements that we expect? I don't see anywhere in these test cases doing what we did above:

```
  when(q.io.deq.fire()) {
    //ensure that what comes out is what comes in
    assert(elems(outCnt.value) === q.io.deq.bits)
    outCnt.inc()
  }
```

in the presence of flushes. this seems like a pretty big test hole.

This would also help demonstrate the expected behavior because you'd have to write the logic to demonstrate the expected behavior of enqueues and dequeues as the same time as flushes (I still don't see any tests that demonstrate that -- specifically directed tests for "i can still enqueue into a non-full queue while flushing, but I don't expect to read what I enqueued" and "I can dequeue from a non-empty queue while flushing, and I expect to read what I enqueued".